### PR TITLE
refactor: Import index files explicitly to maximize compatibility with module loaders

### PR DESCRIPTION
**Pull request by jpray**
This PR addresses https://github.com/material-components/material-components-web/issues/1991

### CODE REVIEW

**Comment by jpray** [comment]
I signed it!

**Comment by jpray** [comment]
Thanks for catching that.  I've made the update.

File: scripts/check-imports.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ let exitCode = 0;
 main();
 
 function main() {
-  globSync('packages/mdc-*/**/*.js')
+  globSync('packages/**/*.js')
```
Comment (bonniezhou) [inline scripts/check-imports.js]: Was there a reason that this was changed?

Comment (jpray) [reply scripts/check-imports.js]: Yes. That glob wasn't including packages/material-components-web/index.js, and all of these checks for imports should apply there as well.

File: scripts/check-imports.js [hunk 2]
```diff
@@ -87,6 +88,30 @@ function assertImportResolvesWithinPkgRoot(source, pkgRoot, srcFile, node) {
   }
 }
 
+function assertNoLocalDirectoryImports(source, pkgRoot, srcFile, node) {
```
Comment (kfranqueiro) [inline scripts/check-imports.js]: Thanks for adding this, btw :star:

Comment (jpray) [reply scripts/check-imports.js]: My pleasure.  Thanks for blessing it and merging it : )

**Pull request merged**